llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -566,14 +570,33 @@ def _get_hf_model(self, state: State):
         log.debug('Gathering state dict')
 
         if state.is_model_ddp:
+            # Type assertions for DDP case
+            assert isinstance(state.model, HuggingFaceModel)
+            assert isinstance(state.model.module, HuggingFaceModel)
+            assert isinstance(state.model.module.model, PreTrainedModel)
+            assert isinstance(
+                state.model.module.tokenizer,
+                PreTrainedTokenizerBase,
+            )
+
             original_model: PreTrainedModel = state.model.module.model
             state_dict_model = state.model.module.model
             original_tokenizer = state.model.module.tokenizer
         elif isinstance(state.model.model, FSDP):
+            # Type assertions for FSDP case
+            assert isinstance(state.model, HuggingFaceModel)
+            assert isinstance(state.model.model.module, PreTrainedModel)
+            assert isinstance(state.model.tokenizer, PreTrainedTokenizerBase)
+
             original_model: PreTrainedModel = state.model.model.module
-            state_dict_model = state.model.model
+            state_dict_model = state.model.model.module
```

Comment:
are you sure this is right? did this change?

Reply:
oops that might be an autocorrect issue, thanks for catching that!